src/ZODB/tests/testMVCCMappingStorage.py

```diff
@@ -33,6 +33,15 @@
     )
 
 class MVCCTests:
+    def checkClosingNestedDatabasesWorks(self):
+        # This tests for the error described in
+        # https://github.com/zopefoundation/ZODB/issues/45
+        db1 = DB(self._storage)
+        db2 = DB(None, databases=db1.databases, database_name='2')
+        db1.open().get_connection('2')
```

Comment:
I must admin, I am only triggering observed behavior. From my understanding, the information that db1 has somehow a connection to db2 is lazy added when I try to get the second connection from it.